stablehlo/dialect/ChloOps.td

```diff
@@ -945,4 +945,27 @@ def CHLO_DynamicReshapeOp: CHLO_Op<"dynamic_reshape", [NoSideEffect,
   let results = (outs HLO_Tensor:$result);
 }
 
+def CHLO_CrossReplicaSumOp : CHLO_Op<"cross-replica-sum",
```

Comment:
Now that we have a (somewhat) better idea of what compatibility maintenance may look like (#115), I think it would be better to rename this to `cross_replica_sum`.

We want to eventually rename this because (1) it is the only op that uses hyphens in the mnemonic, and (2) hyphenated ops are not supported by pretty printing.

I think this PR is a good time to do this, there is no compatibility impact until we migrate to start using this CHLO op. And if we did the change separately (`mhlo.cross-replica-sum --> chlo.cross-replica-sum --> chlo.cross_replica_sum`), there are two points of migration that we will need to support for the duration of the forwards/backwards compatibility window.

Doing this all at once also has the advantage of only needing to update test files once, after we start migrating to use this op.